gc_little_helper_II.user.js

```diff
@@ -4886,6 +4887,181 @@ var mainGC = function() {
+            let px = getURLParam('gclh_px');
+            let radius = getRadius(lat, zoom, px);
+            // Cache Types.
+            let cacheTypes = getURLParam('ct') ? decodeURIComponent(getURLParam('ct')).split(',') : false;
+            // Found Status.
+            let showFound = getURLParam('hf') && getURLParam('hf') == 0 ? true : false;
+            let hideFound = getURLParam('hf') && getURLParam('hf') == 1 ? true : false;
+            // Hide Status.
+            let showOwn = getURLParam('ho') && getURLParam('ho') == 0 ? true : false;
+            let hideOwn = getURLParam('ho') && getURLParam('ho') == 1 ? true : false;
+            // Difficult and terrin rating.
+            let d_t = getTDMinMax();
+            // Cache Size.
+            let cacheSize = getURLParam('cs') ? decodeURIComponent(getURLParam('cs')).split(',') : false;
+            // Membership type.
+            let basic = getURLParam('sp') && getURLParam('sp') == 0 ? true : false;
+            let premium = getURLParam('sp') && getURLParam('sp') == 1 ? true : false;
+            // Cache Status.
+            let enabled = getURLParam('sd') && getURLParam('sd') == 0 ? true : false;
+            let disabled = getURLParam('sd') && getURLParam('sd') == 1 ? true : false;
+            // Placed Date.
+            let placedDateStart = false;
+            if (getURLParam('pad')) placedDateStart = getURLParam('pad');
+            if (getURLParam('psd')) placedDateStart = getURLParam('psd');
+            if (getURLParam('pod')) placedDateStart = getURLParam('pod');
+            let placedDateEnd = false;
+            if (getURLParam('pbd')) placedDateEnd = getURLParam('pbd');
+            if (getURLParam('ped')) placedDateEnd = getURLParam('ped');
+            if (getURLParam('pod')) placedDateEnd = getURLParam('pod');
+            // Attribute
+            let attr = getURLParam('att') ? decodeURIComponent(getURLParam('att')).split(',') : false;
+
+            // Warning if filters are not available at PQ.
+            let warningHTML = '<div id="gclh_warning" style="color:#f00"><span>Warning: You have set filters that cannot be implemented in Pocket Queries.<br>This applies to the following filters:</span><ul></ul></div>'
+            if (getURLParam('tr') || getURLParam('sgt') || (getURLParam('nfb') && getURLParam('nfb') != global_me) || (getURLParam('hb') && getURLParam('hb') != global_me) || getURLParam('cc') || getURLParam('cn') || getURLParam('fp')) {
+                $('#ctl00_ContentBody_QueryPanel').before(warningHTML);
+                let unavailableFilters = '';
+                if (getURLParam('tr')) unavailableFilters += '<li>Filter by Wonder</li>';
+                if (getURLParam('sgt')) unavailableFilters += '<li>Only show caches that are part of a GeoTour</li>';
+                if (getURLParam('nfb')) unavailableFilters += '<li>Not found by</li>';
+                if (getURLParam('hb')) unavailableFilters += '<li>Hidden by</li>';
```

Comment:
Ich habe hier mal etwas getestet. Sieht ja schon ganz gut aus.

Die Logik oben passt zumindest dann nicht wenn mehr als ein Filter nicht verfügbar ist. Beispielsweise Wonder Gedöhns und Not Found bei me. Das führt dann dazu dass beide Fehler ausgegeben werden. Du musst jeden Filter einzeln prüfen und dann eine Fehlermeldung ausgeben.

Reply:
Es werden doch alle Filter einzeln geprüft und dann alle oben gebündelt als Liste ausgegeben.

Reply:
Aber 'Not found by me' ist doch doch gar kein Fehler. 

Man kommt nur in die Ausgabe weil 'Wonder Gedöhns' ein Fehler ist. Und dann wird 'Wonder Gedöhns' als Fehler ausgegeben (OK), aber 'Not Found by me' eben auch (Nicht OK). Letzteres nur deshalb, weil das 'nfb' vorhanden ist, obwohl es ja 'Not found by me' ist und nicht 'Not found by jemand anderem'. 

Also: Einzeln prüfen und dann ausgeben. 

Probier es aus mit den beiden Filtern.

Reply:
Ah, das meinst du. 
Das werde ich anpassen.

Reply:
So, oben hatte ich es richtig gemacht, aber dann beim einzelnen Prüfen nicht... jetzt sollte es klappen